src/main/java/org/sonatype/goodies/packageurl/PackageUrlBuilder.java

```diff
@@ -135,7 +139,17 @@ public PackageUrlBuilder subpath(final String subpath) {
    * At minimal {@link #type} and {@link #name} must be specified.
    */
   public PackageUrl build() {
-    return buildAndValidate(true);
+    return buildAndValidate(true, true);
+  }
+
+  /**
+   * Build a {@link PackageUrl} "as is".
+   * That is, will *not* make the corrections defined in the spec to the namespace, to the name or to the qualifier keys.
+   *
+   * At minimal {@link #type} and {@link #name} must be specified.
+   */
+  public PackageUrl buildAsIs() {
+    return buildAndValidate(true, false);
```

Comment:
I recommend we don't add special api here, and instead just put a flag on the builder and expose the ability to set it

Reply:
Done